kernel/ota_update_client/src/lib.rs

```diff
@@ -324,11 +319,11 @@ fn download_files<S: AsRef<str>>(
     }
 
     let startup_time = hpet_ticks!();
-    let mut rng = SmallRng::seed_from_u64(startup_time);
     let rng_upper_bound = u16::max_value() - STARTING_FREE_PORT;
 
     // the below items may be overwritten on each loop iteration, if the socket was closed and we need to create a new one
-    let mut local_port = STARTING_FREE_PORT + (rng.next_u32() as u16 % rng_upper_bound);
+    // TODO: This doesn't provide a uniform distribution of port numbers.
```

Comment:
this TODO is interesting, can you elaborate? Are you saying that `random::next_u32()` doesn't generate truly random numbers?

Reply:
Take a particularly egregious example:
```rust
/// Generates a random number between 0-20.
fn gen_random() -> u8 {
    // generate
}

let x = gen_random();
let y = x % 15;
```
`y` is as likely to be in the range 0-4 as it is to be in the range 5-14.

Looking over the code again, removing the cast to `u16` would fix this as the difference in probability would be incredibly small.

Reply:
ok yeah, so to fix it we just want to truncate the `rng.next_u32()` value after we modulo it with the upper bound, not before.

Reply:
we'll do that in the future PR that changes other crates to use separate PRNGs based on our new singleton CSPRNG